include/mbgl/shaders/heatmap_layer_ubo.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <mbgl/shaders/layer_ubo.hpp>
+
+namespace mbgl {
+namespace shaders {
+
+struct alignas(16) HeatmapDrawableUBO {
+    std::array<float, 4 * 4> matrix;
+    float extrude_scale;
+    std::array<float, 3> padding;
```

Comment:
Won't a 12-byte array align on a 16-byte boundary?

Reply:
From what I know this happens only in GLSL.

Reply:
It would in GLSL, but only if declared using `vec3` in the GLSL. If you declared `extrude_scale` and `padding` together as a `vec4`, it wouldn't.